### DISCUSSION TITLE
Testers/developers needed for new  -development-

### DISCUSSION THREAD

**Discussion by TomTravis** [dnschneid/crouton#4891] [General]
My repo is not for maintenance only

I have made some forward motion with Crouton and systemd 

Github gave us test instructions they cannot do the verification via the website

if interested write to the owner and myself

Testing starts today

**Comment by TomTravis** [comment]
[crouton.zip](https://github.com/dnschneid/crouton/files/10033669/crouton.zip)
Patching debootstrap...
In file included from src/findnacld.c:5:
src/websocket.h:905:13: warning: ‘socket_server_init’ defined but not used [-Wunused-function]
  905 | static void socket_server_init(int port_) {
      |             ^~~~~~~~~~~~~~~~~~
src/websocket.h:806:12: warning: ‘socket_server_accept’ defined but not used [-Wunused-function]
  806 | static int socket_server_accept(char* version) {
      |            ^~~~~~~~~~~~~~~~~~~~
[mk.txt](https://github.com/dnschneid/crouton/files/10033629/mk.txt)

**Comment by TomTravis** [comment]
![Screenshot 2022-11-17 11 04 01 AM](https://user-images.githubusercontent.com/46386261/202535572-461cff17-5ba6-4d1d-8d89-f992a50869ac.png)
well this is not good news
chronos@localhost / $ sudo test1z -p /media/removable/sd200/z -r focal -t xorg,xiwi,xfce -n foc1
WARNING: focal is an unsupported release.
You will likely run into issues, but things may work with some effort.
Press Ctrl-C to abort; installation will continue in 5 seconds.
Installing focal-arm64 chroot to /media/removable/sd200/z/chroots/foc1
Downloading bootstrap files...
/usr/local/bin/test1z: 95: /tmp/test1z.MHh/installer/ubuntu/bootstrap: /tmp/test1z.Os4/debootstrap: Permission denied
debootstrap error log:
tail: cannot open '/tmp/test1z.Os4/focal-arm64/debootstrap/debootstrap.log' for reading: No such file or directory
Failed to run debootstrap.
after 107 update
now i need to power wash and back to 106 no more autoupdate
edit update works fresh install  does not work

**Comment by TomTravis** [comment]
time to write to google
chronos@localhost ~ $ sudo printf '#!/bin/sh\necho hi\n' > /usr/local/bin/test.sh|tee >>log.txt
bash: /usr/local/bin/test.sh: Permission denied

**Comment by TomTravis** [comment]
https://github.com/chromeos/chromeos.dev/issues/559
issue submitted


the repair now is to post core chroots as tar.gz 
un tar and use  -u which works just fine
this is what go installs to bootstrap

**Comment by TomTravis** [comment]
![Screenshot 2022-11-17 1 22 49 PM](https://user-images.githubusercontent.com/46386261/202562758-4c4b2e8d-2c25-41c5-93fe-b88270949306.png)
so the is the proposed change
touch is required to write via printf to files exec does not matter Printf 'echo' >> file fails now file is not created permisson fails
sudo touch z.sh
sudo chmod 777 z.sh
sudo printf '#!/bin/sh\necho hi\n' >>z.sh
z.sh
works
now to edit the code
![Screenshot 2022-11-17 7 17 16 PM](https://user-images.githubusercontent.com/46386261/202612900-4e1ff5cb-f7bc-4b65-bcd8-a63e897d9445.png)
![Screenshot 2022-11-17 7 27 59 PM](https://user-images.githubusercontent.com/46386261/202612918-c1f79b01-93f1-4c8b-a64f-19de679e2705.png)

**Comment by TomTravis** [comment]
this is code that fails due to chromeOS undocumented changes
the cp fails and the tail .log fails

    # Add the necessary debootstrap executables
    newpath="$PATH:$tmp"
    echo "cp fail" have to test
    cp "$INSTALLERDIR/$DISTRO/ar" "$INSTALLERDIR/$DISTRO/pkgdetails" "$tmp/"
    chmod 755 "$tmp/ar" "$tmp/pkgdetails"

    # Grab the release and drop it into the subdirectory

    echo 'Downloading bootstrap files...' 1>&2

    if ! PATH="$newpath" DEBOOTSTRAP_DIR="$tmp" $FAKEROOT \

No trap for error handle Great care is taken in over parts of the code for error handle

            **"$tmp/debootstrap" --foreign --extractor='ar' --arch="$ARCH" \** # does not run

            "$RELEASE" "$tmp/$subdir" "$MIRROR" 1>&2; then

            echo "echo log fail"

        echo "debootstrap error log:" 1>&2

        tail -n 3 "$tmp/$subdir/debootstrap/debootstrap.log" 1>&2 || true

        error 1 'Failed to run debootstrap.'
    fi
fi

**Comment by TomTravis** [comment]
issue 
https://github.com/dnschneid/crouton/issues/4884
and
https://github.com/chromeos/chromeos.dev/issues/559

**Reply by TomTravis** [reply]
[log.txt](https://github.com/dnschneid/crouton/files/10045602/log.txt)

**Comment by TomTravis** [comment]
![Screenshot 2022-11-18 3 38 48 PM](https://user-images.githubusercontent.com/46386261/202822911-632f2c8c-d3c3-47ec-ba18-d0f00871c5fc.png)
![Screenshot 2022-11-18 3 40 53 PM](https://user-images.githubusercontent.com/46386261/202822916-cf40ad51-2fe9-48d0-a351-d61cfeae9fbd.png)
![Screenshot 2022-11-18 3 45 04 PM](https://user-images.githubusercontent.com/46386261/202822946-a5320926-0dd1-4ffc-a1b6-c56fb4cafd0b.png)

**Comment by TomTravis** [comment]
try this was change to run in /usr/local/bin
https://drive.google.com/uc?export=download&id=1HFjX1OejjHAYbaHVL3l9krmBUkA1caZR
test2 same a crouton -r -t
this may fix the permission issue
seem to work for me

**Reply by TomTravis** [reply]
PenguinTest is now work great news test3
![Screenshot 2022-11-19 4 32 39 PM](https://user-images.githubusercontent.com/46386261/202877438-8c9403c5-3073-4eab-a481-ff37466dc6df.png)

**Comment by bsammon** [comment]
I'd be interested in trying this.  I just recently acquired a Chromebook.  It's got ChromeOS 107 on it.
Is it a good match for this?
What would I do to test this?
